Fix a faulty check which resulted in ZLIB being required on all platforms.

Author: Alexander Grund (TU Dresden)

diff --git a/src/cpp/ext/CMakeLists.txt b/src/cpp/ext/CMakeLists.txt
--- a/src/cpp/ext/CMakeLists.txt
+++ b/src/cpp/ext/CMakeLists.txt
@@ -55,9 +65,10 @@ function(dependency)
    set(${_NAME}_PLATFORMS  "${_PLATFORMS}"  CACHE INTERNAL "")
    set(${_NAME}_CXXFLAGS   "${_CXXFLAGS}"   CACHE INTERNAL "")
 
+   set(${_NAME}_ENABLED FALSE)
    if(_PLATFORMS)
       foreach(_PLATFORM IN LISTS _PLATFORMS)
-         if(_PLATFORM)
+         if(${_PLATFORM})
             set(${_NAME}_ENABLED TRUE)
             break()
          endif()
